src/cargo/core/source/source_id.rs

```diff
@@ -680,7 +680,13 @@ impl<'a> fmt::Display for SourceIdAsUrl<'a> {
                 kind: SourceKind::Registry,
                 ref url,
                 ..
-            } => write!(f, "registry+{}", url),
+            } => {
+                if url.scheme().starts_with("sparse+") {
+                    write!(f, "{}", url)
+                } else {
+                    write!(f, "registry+{}", url)
+                }
+            }
```

Comment:
Is this needed for the PR or an incidental artifact of being split out of another PR?

If its needed, should it be split out into a separate commit with a description of why this is needed?

Reply:
It's not needed. I'll split it out into another PR and add a test.